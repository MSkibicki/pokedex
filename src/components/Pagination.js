import React from "react";

const Pagination = ({ handlePreviousPage, handleNextPage }) => {
  return (
    <div>
      {handlePreviousPage && <button onClick={handlePreviousPage}>Previous Page</button>}
      {handleNextPage && <button onClick={handleNextPage}>Next Page</button>}
    </div>
  );
};

export default Pagination;
