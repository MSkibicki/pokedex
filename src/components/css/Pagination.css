.buttons {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
      -ms-flex-pack: center;
          justify-content: center;
  margin-bottom: 3em;
  margin-top: 1em;
}

.buttons .button-previous {
  height: 40px;
  width: 150px;
  border-radius: 10px;
  margin-right: 1em;
  background: linear-gradient(120deg, #fff, #000);
  color: #f2f7eb;
}

.buttons .button-next {
  height: 40px;
  width: 150px;
  border-radius: 10px;
  margin-left: 1em;
  background: linear-gradient(120deg, #fff, #000);
  color: #f2f7eb;
}

.buttons .button-previous:hover {
  cursor: pointer;
  -webkit-box-shadow: 5px 5px 5px black;
          box-shadow: 5px 5px 5px black;
  outline: 0;
}

.buttons .button-next:hover {
  cursor: pointer;
  -webkit-box-shadow: 5px 5px 5px black;
          box-shadow: 5px 5px 5px black;
  outline: 0;
}
/*# sourceMappingURL=Pagination.css.map */