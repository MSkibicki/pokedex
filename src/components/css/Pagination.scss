@import "./Global.scss";

.buttons {
  display: flex;
  justify-content: center;
  margin-bottom: 3em;
  margin-top: 1em;

  .button-previous {
    height: 40px;
    width: 150px;
    border-radius: 10px;
    margin-right: 1em;
    background: linear-gradient(120deg, #fff, $black);
    color: $main-color;
  }

  .button-next {
    height: 40px;
    width: 150px;
    border-radius: 10px;
    margin-left: 1em;
    background: linear-gradient(120deg, #fff, $black);
    color: $main-color;
  }

  .button-previous:hover {
    cursor: pointer;
    box-shadow: 5px 5px 5px $black;
    outline: 0;
  }

  .button-next:hover {
    cursor: pointer;
    box-shadow: 5px 5px 5px $black;
    outline: 0;
  }
}
