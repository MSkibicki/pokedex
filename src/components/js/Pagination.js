import React from "react";
import "../css/Pagination.scss";
import PropTypes from "prop-types";

const Pagination = ({ handlePreviousPage, handleNextPage }) => {
  return (
    <div className="buttons">
      {handlePreviousPage && (
        <button className="button-previous" onClick={handlePreviousPage}>
          Previous Page
        </button>
      )}
      {handleNextPage && (
        <button className="button-next" onClick={handleNextPage}>
          Next Page
        </button>
      )}
    </div>
  );
};

Pagination.propTypes = {
  handlePreviousPage: PropTypes.func,
  handleNextPage: PropTypes.func,
};

export default Pagination;
