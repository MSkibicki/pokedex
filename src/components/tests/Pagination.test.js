import Pagination from "./../js/Pagination";

describe("<Pagination />", () => {
  it("renders correctly", () => {
    const wrapper = shallow(
      <Pagination handlePreviousPage={() => {}} handleNextPage={() => {}} />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it("displays previous button text correctly", () => {
    const wrapper = mount(
      <Pagination handlePreviousPage={() => {}} handleNextPage={() => {}} />
    );

    const text = wrapper.find(".button-previous").text();

    expect(text).toEqual("Previous Page");
  });

  it("displays next button text correctly", () => {
    const wrapper = mount(
      <Pagination handlePreviousPage={() => {}} handleNextPage={() => {}} />
    );

    const text = wrapper.find(".button-next").text();

    expect(text).toEqual("Next Page");
  });

  it("calls handlePreviousPage on click", () => {
    const spy = sinon.spy();
    const wrapper = mount(
      <Pagination handlePreviousPage={spy} handleNextPage={() => {}} />
    );

    wrapper.find(".button-previous").simulate("click");

    expect(spy.calledOnce).toBe(true);
  });

  it("calls handleNextPage on click", () => {
    const spy = sinon.spy();
    const wrapper = mount(
      <Pagination handlePreviousPage={() => {}} handleNextPage={spy} />
    );

    wrapper.find(".button-next").simulate("click");

    expect(spy.calledOnce).toBe(true);
  });
});
