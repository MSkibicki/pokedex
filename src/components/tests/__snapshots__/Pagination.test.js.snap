// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="buttons"
>
  <button
    className="button-previous"
    onClick={[Function]}
  >
    Previous Page
  </button>
  <button
    className="button-next"
    onClick={[Function]}
  >
    Next Page
  </button>
</div>
`;
